### PR TITLE
Assigned emptyDir for etcd data dir in skydns pod

### PR DESCRIPTION
**Pull request by ArtfulCoder**
@thockin

### CODE REVIEW

**Comment by ArtfulCoder** [comment]
@bprashanth @dchen1107 
issue #10926

**Comment by ArtfulCoder** [comment]
The 3 tests that are failing in this e2e are failing already in all the k8s bot runs for other in-flight PRs.
So I believe this to be un-related to my changes and should be safe to merge.

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 1]
```diff
@@ -28,12 +28,17 @@ spec:
             memory: 50Mi
         command:
         - /usr/local/bin/etcd
+        - -data-dir
+        - /var/etcd/data
```
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: this is a subdir of the volumeMount - is that intentional?

**Comment by thockin** [comment]
LGTM

**Comment by thockin** [comment]
@jnagal for priority merge once @bprashanth OKs

File: cluster/addons/dns/skydns-rc.yaml.in [hunk 2]
```diff
@@ -28,12 +28,17 @@ spec:
...
         - -listen-client-urls
         - http://127.0.0.1:2379,http://127.0.0.1:4001
         - -advertise-client-urls
         - http://127.0.0.1:2379,http://127.0.0.1:4001
         - -initial-cluster-token
         - skydns-etcd
+        volumeMounts:
```
Comment (bprashanth) [inline cluster/addons/dns/skydns-rc.yaml.in]: optional nit: etcd logs would also be nice

**Comment by bprashanth** [comment]
LGTM sans the volumeMount point. Also can you please wait for https://github.com/GoogleCloudPlatform/kubernetes/pull/11004 and rebase? :)

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 1 above)
Comment (ArtfulCoder) [inline cluster/addons/dns/skydns-rc.yaml.in]: I have now mounted mounted emptydir on /var/etcd/data 
I used that as a mount mount for consistency. Our other etcd container on master uses /var/etcd/data

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 2 above)
Comment (ArtfulCoder) [inline cluster/addons/dns/skydns-rc.yaml.in]: I will have to wrap the etcd command in a shell for stdout and stderr to be redirected. I can do it in a separate PR

**Comment by thockin** [comment]
marked ok to merge

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 2 above)
Comment (thockin) [inline cluster/addons/dns/skydns-rc.yaml.in]: don't redirect stdout/err - then 'kubectl logs` won't work

**Comment by thockin** [comment]
@rjnagal

**Comment by rjnagal** [comment]
#11004 merged. Can you rebase?

**Comment by ArtfulCoder** [comment]
@rjnagal rebased

**Comment by thockin** [comment]
@artfulcoder ASAP please

**Comment by thockin** [comment]
LGTM still, thanks

**Comment by rjnagal** [comment]
merging

File: cluster/addons/dns/skydns-rc.yaml.in (hunk 2 above)
Comment (bprashanth) [inline cluster/addons/dns/skydns-rc.yaml.in]: I was just asking for -debug https://github.com/coreos/etcd/blob/master/Documentation/configuration.md, but it looks like it isn't available in etcd 2.0 :\

**Comment by ArtfulCoder** [comment]
Rohit is waiting for  e2e

**Comment by rjnagal** [comment]
passed and merged.

**Pull request merged**